core/model/objects/asset.cpp

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright Soramitsu Co., Ltd. 2016 All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "asset.hpp"
+
+namespace asset {
+
+    Asset::Asset(
+            const std::string domain,
+            const std::string name,
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function parameter 'name' should be passed by reference.](https://www.codacy.com/app/soramitsu/iroha/file/4345715879/issues/source?bid=3712491&fileBranchId=3870389#l23)